File: lessons/compiled_lang/cs/src/o10.options.cs
using System;
using System.Collections.Generic;

class O10Options
{
    static void Usage(System.IO.TextWriter outp, string scriptName)
    {
        outp.WriteLine();
        outp.WriteLine("Usage: " + scriptName + " [-c] [-e] [-l] [-k] [-p] [-m] [-t] [-h|-?]");
        outp.WriteLine();
        outp.WriteLine("  -c  Coffee");
        outp.WriteLine("  -e  Espresso");
        outp.WriteLine("  -l  Latte");
        outp.WriteLine("  -k  Machiato");
        outp.WriteLine("  -p  Capucino");
        outp.WriteLine("  -m  Mocha");
        outp.WriteLine("  -t  Tea");
        outp.WriteLine("  -h  Display this help message");
        outp.WriteLine("  -?  Display this help message");
        outp.WriteLine();
    }

    static void Main(string[] args)
    {
        string scriptName = Environment.GetCommandLineArgs()[0];

        if (args.Length == 0)
        {
            Usage(Console.Error, scriptName);
            Environment.Exit(1);
            return;
        }

        List<string> orders = new List<string>();

        foreach (string arg in args)
        {
            switch (arg)
            {
                case "-c":
                    orders.Add("coffee");
                    break;
                case "-e":
                    orders.Add("espresso");
                    break;
                case "-l":
                    orders.Add("latte");
                    break;
                case "-k":
                    orders.Add("macchiato");
                    break;
                case "-p":
                    orders.Add("capucino");
                    break;
                case "-m":
                    orders.Add("mocha");
                    break;
                case "-t":
                    orders.Add("tea");
                    break;
                case "-h":
                case "-?":
                    Usage(Console.Out, scriptName);
                    Environment.Exit(0);
                    return;
                default:
                    Usage(Console.Error, scriptName);
                    Environment.Exit(1);
                    return;
            }
        }

        Console.WriteLine();
        Console.WriteLine("You ordered: ");
        foreach (string drink in orders)
        {
            Console.WriteLine("* " + drink);
        }
    }
}

File: lessons/compiled_lang/java/src/o10.options.java
import java.io.PrintStream;
import java.util.ArrayList;
import java.util.List;

// Not public - see a00.output.java for why.
class O10Options {
    private static void usage(PrintStream out, String scriptName) {
        out.println();
        out.println("Usage: " + scriptName + " [-c] [-e] [-l] [-k] [-p] [-m] [-t] [-h|-?]");
        out.println();
        out.println("  -c  Coffee");
        out.println("  -e  Espresso");
        out.println("  -l  Latte");
        out.println("  -k  Machiato");
        out.println("  -p  Capucino");
        out.println("  -m  Mocha");
        out.println("  -t  Tea");
        out.println("  -h  Display this help message");
        out.println("  -?  Display this help message");
        out.println();
    }

    public static void main(String[] args) {
        String scriptName = System.getProperty("invoked.as");

        if (args.length == 0) {
            usage(System.err, scriptName);
            System.exit(1);
            return;
        }

        List<String> orders = new ArrayList<>();
        for (String arg : args) {
            switch (arg) {
                case "-c": orders.add("coffee"); break;
                case "-e": orders.add("espresso"); break;
                case "-l": orders.add("latte"); break;
                case "-k": orders.add("macchiato"); break;
                case "-p": orders.add("capucino"); break;
                case "-m": orders.add("mocha"); break;
                case "-t": orders.add("tea"); break;
                case "-h":
                case "-?":
                    usage(System.out, scriptName);
                    return;
                default:
                    usage(System.err, scriptName);
                    System.exit(1);
                    return;
            }
        }

        System.out.println();
        System.out.println("You ordered: ");
        for (String order : orders) {
            System.out.println("* " + order);
        }
    }
}
